shared/src/business/useCaseHelper/caseAssociation/createUserForContactPrimary.test.js

```diff
@@ -0,0 +1,141 @@
+const {
+  applicationContext,
+} = require('../../test/createTestApplicationContext');
+const {
+  createUserForContactPrimary,
+} = require('./createUserForContactPrimary');
+const {
+  ROLES,
+  SERVICE_INDICATOR_TYPES,
+} = require('../../entities/EntityConstants');
+const { Case } = require('../../entities/cases/Case');
+const { MOCK_CASE } = require('../../../test/mockCase');
+
+describe('createUserForContactPrimary', () => {
+  const USER_ID = '674fdded-1d17-4081-b9fa-950abc677cee';
+
+  beforeEach(() => {
+    applicationContext.getUniqueId.mockReturnValue(USER_ID);
+  });
+
+  it('should throw an unauthorized error for non admissionsclerk users', async () => {
+    applicationContext.getCurrentUser.mockReturnValue({});
+
+    await expect(
+      createUserForContactPrimary({
+        applicationContext,
+        caseEntity: new Case(MOCK_CASE, { applicationContext }),
+        email: 'testing@example.com',
+        name: 'Bob Ross',
+      }),
+    ).rejects.toThrow('Unauthorized');
+  });
+
+  it('should call createNewPetitionerUser with the new user entity', async () => {
+    const UPDATED_EMAIL = 'testing@example.com';
+
+    applicationContext.getCurrentUser.mockReturnValue({
+      role: ROLES.admissionsClerk,
+    });
+
+    const caseEntity = new Case(
+      {
+        ...MOCK_CASE,
+        contactPrimary: {
+          ...MOCK_CASE.contactPrimary,
+          contactId: USER_ID,
+          email: undefined,
+          name: 'Bob Ross',
+          serviceIndicator: SERVICE_INDICATOR_TYPES.SI_PAPER,
+        },
+      },
+      { applicationContext },
+    );
+
+    await createUserForContactPrimary({
+      applicationContext,
+      caseEntity,
+      email: UPDATED_EMAIL,
+      name: 'Bob Ross',
+    });
+
+    expect(
+      applicationContext.getPersistenceGateway().createNewPetitionerUser.mock
+        .calls[0][0].user,
+    ).toMatchObject({
+      contact: {},
+      name: 'Bob Ross',
+      pendingEmail: UPDATED_EMAIL,
+      role: ROLES.petitioner,
+      userId: USER_ID,
+    });
+  });
+
+  it('should return the caseEntity', async () => {
+    const UPDATED_EMAIL = 'testing@example.com';
+
+    applicationContext.getCurrentUser.mockReturnValue({
+      role: ROLES.admissionsClerk,
+    });
+
+    const caseEntity = new Case(
+      {
+        ...MOCK_CASE,
+        contactPrimary: {
+          ...MOCK_CASE.contactPrimary,
+          contactId: USER_ID,
+          email: undefined,
+          name: 'Bob Ross',
+          serviceIndicator: SERVICE_INDICATOR_TYPES.SI_PAPER,
+        },
+      },
+      { applicationContext },
+    );
+
+    const updatedCase = await createUserForContactPrimary({
+      applicationContext,
+      caseEntity,
+      email: UPDATED_EMAIL,
+      name: 'Bob Ross',
+    });
+
+    expect(updatedCase).toMatchObject(caseEntity);
```

Comment:
Wouldn't this always be true? What is this trying to test? `createUserForContactPrimary` does not alter the case. Even if it did, it would mutate the object we passed in and this would always pass. 

Makes me think that the original function should receive what only it needs:

* docketNumber
* contactPrimary

... and not worry about returning something it doesn't update

Reply:
The intent of the test is to prove that the caseEntity that was given is returned - WITHOUT mutation. If you look at places where this method is called, a case entity is instantiated and mutated, this data is then used in THIS method to create the user record, and then handed back to the calling method to return this case in the response.

See `updatePetitionerInformationInteractor` for an example of this.

Reply:
The thing is, if you actually modify the code to mutate the case, the test still passes.